RELEASE.md

```diff
@@ -110,7 +110,7 @@ Now assume `x.y.z` is the new version.
 
     ```
     git pull
-    git checkout master
+    git checkout release/x.y.z
```

Comment:
Note, this assumes the existence of a release/x.y.z branch; this is something not mandatory right now per release document in Falco.
We might want to evaluate to follow the same approach that was taken on libs, ie forcing a `release/M.m.x` release branch naming (https://github.com/falcosecurity/libs/pull/821), and then eventually cherry pick on the branch if more changes are needed.
We already got tags that point to a given commit hash (ie: the release one) anyway.

This would also help us cleanup protected branch on test-infra configs that rn are a bit of a mess: https://github.com/falcosecurity/test-infra/blob/master/config/config.yaml#L152

(it would become something like:
```
branches:
            master:
              protect: true
            "release/0.28.1":
              protect: true
            "release/0.32.2":
              protect: true
            "release/0.33.0":
              protect: true
            "release/0.33.1":
              protect: true
            "release/0.34.x":
              protect: true
             "release/0.35.x":
              protect: true
...
```

Reply:
I totally agree to use the same branching strategy we are using in libs (ie. `release/M.m.x`).